src/Microsoft.AspNet.Security.OAuthBearer/OAuthBearerAuthenticationHandler.cs

```diff
@@ -0,0 +1,195 @@
+                        AuthenticationTicket ticket = new AuthenticationTicket(principal, new AuthenticationProperties(), Options.AuthenticationType);
+                        var securityTokenValidatedNotification = new SecurityTokenValidatedNotification<HttpContext, OAuthBearerAuthenticationOptions>(Context, Options)
+                        {
+                            ProtocolMessage = Context,
+                            AuthenticationTicket = ticket
+                        };
+
+                        if (securityTokenReceivedNotification.HandledResponse)
+                        {
+                            return securityTokenValidatedNotification.AuthenticationTicket;
+                        }
+
+                        if (securityTokenReceivedNotification.Skipped)
+                        {
+                            return null;
+                        }
+
+                        return ticket;
+                    }
+                }
+
+                throw new InvalidOperationException("No SecurityTokenValidator available for token: " + token ?? "null");
+            }
+            catch (Exception ex)
+            {
+                // We can't await inside a catch block, capture and handle outside.
+                authFailedEx = ExceptionDispatchInfo.Capture(ex);
+            }
+
+            if (authFailedEx != null)
+            {
+                _logger.WriteError("Exception occurred while processing message: '" + authFailedEx.ToString());
+
+                // Refresh the configuration for exceptions that may be caused by key rollovers. The user can also request a refresh in the notification.
+                if (Options.RefreshOnIssuerKeyNotFound && authFailedEx.SourceException.GetType().Equals(typeof(SecurityTokenSignatureKeyNotFoundException)))
+                {
+                    Options.ConfigurationManager.RequestRefresh();
+                }
+
+                var authenticationFailedNotification =
+                    new AuthenticationFailedNotification<HttpContext, OAuthBearerAuthenticationOptions>(Context, Options)
+                    {
+                        ProtocolMessage = Context,
+                        Exception = authFailedEx.SourceException
+                    };
+
+                await Options.Notifications.AuthenticationFailed(authenticationFailedNotification);
+                if (authenticationFailedNotification.HandledResponse)
+                {
+                    return authenticationFailedNotification.AuthenticationTicket;
+                }
+
+                if (authenticationFailedNotification.Skipped)
+                {
+                    return null;
+                }
+
+                authFailedEx.Throw();
+            }
+
+            return null;
+        }
+
+        protected override void ApplyResponseChallenge()
+        {
+            // N/A
```

Comment:
Bearer does need to issue a challenge. the default is just `Bearer`, but there should be a notification extension.